### DISCUSSION TITLE
Forgot how to export and import

### DISCUSSION THREAD

**Discussion by GH-superadmin** [netbox-community/netbox#17030] [General]
I cannot remember or find the method to export and import to create a new device template.

I have some ESXi servers I have created and I expanded the interfaces for one of them to include the VMkernals and VMnics etc.

I would like to export that server and import it as a new device template.

Any help is appreciated.

Thanks

**Comment by candlerb** [comment]
There isn't really an out-of-the-box way to do that.

You can export the interfaces by going to `/dcim/interfaces/`, then setting a filter on a single device, then Export > All Data.  But what you'll get is a CSV file, with one row per interface. A device type export/import is in YAML format, with a nested list of interfaces, and it would be up to you to assemble the YAML in the correct format.

You can find examples of the YAML for device types in the [devicetype library](https://github.com/netbox-community/devicetype-library/tree/master/device-types).
